xtensa-build-all: document PRIVATE_KEY_OPTION and deprecate -k

marc-hb opened the pull request:
Following commit 88b69cc2e5ca ("scripts: xtensa-build-all: Add support
for building TGL and signing with MEU") we now have an unified and
flexible PRIVATE_KEY_OPTION that can do anything with very little code.

The -k option was never a good idea in the first place because it
already required an environment variable anyway (RIMAGE_PRIVATE_KEY)
instead of a command line argument so the code should have just checked
whether RIMAGE_PRIVATE_KEY was defined, that would have been
enough. Requiring the user to "double-confirm" with -k has been adding
extra complication for both the user interface and the implementation.

xtensa-build-all is effectively a "CMake configuration convenience"
script, however it shouldn't become an additional layer of indirection
and complexity and its interface should expose CMake (which is already a
layer of indirection!) as directly as possible.

See longer discussion in PR #3187

Signed-off-by: Marc Herbert <marc.herbert@intel.com>

juimonen commented:
@marc-hb, yes looks good.

I wonder could we even add a concrete example like in some man pages, like:

to build with docker for tgl with private/external key and meu tool:
export PRIVATE_KEY_OPTION=-DMEU_PRIVATE_KEY=/path_to_key/private.pem
./scripts/docker-run.sh ./scripts/xtensa-build-all.sh -m /path_to_meu/ tgl

This would make it very clear how to do it.

lgirdwood commented on scripts/xtensa-build-all.sh:
```diff
@@ -113,6 +125,7 @@ fi
 
 if [ "x$USE_PRIVATE_KEY" == "xyes" ]
 then
+	>&2 printf 'WARNING: -k and RIMAGE_PRIVATE_KEY are deprecated\n'
```
"...use -new-switch or new ENV instead"

marc-hb replied:
To avoid duplication I added "see usage".

lgirdwood reviewed (commented):
looks good, could you add a litte more help for the user so they know the new switch/env variable to use in this case.

marc-hb commented:
> I wonder could we even add a concrete example like in some man pages,

We don't have man pages, I think the closest is sof-docs. Happy to review anything you submit there :-)

> like, to build with docker for tgl with private/external key and meu tool:
```
export PRIVATE_KEY_OPTION=-DMEU_PRIVATE_KEY=/path_to_key/private.pem
./scripts/docker-run.sh ./scripts/xtensa-build-all.sh -m /path_to_meu/ tgl
```
I already added something very close to this in the `usage`. I just added the `-m /path_to_meu/` now.

I didn't add `tgl` because I prefer to keep the example generic. The platform name is the most basic parameter that everyone has been using since forever _and_ it's the first line of the `usage` message that gets printed if you forget it.

I didn't add `./scripts/docker-run.sh` either because this script is unrelated to docker. I use this script all the time and I never use docker.
